controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -465,10 +466,16 @@ func (pp *portPublisher) endpointsToAddresses(endpoints *corev1.Endpoints) Addre
 					Namespace: endpoints.ObjectMeta.Namespace,
 				}
 
+				var authorityOverride string
+				if fqName, ok := endpoints.Annotations[consts.RemoteServiceFqName]; ok {
+					authorityOverride = fmt.Sprintf("%s:%d", fqName, resolvedPort)
```

Comment:
is this port correct?  I think `resolvedPort` is the gateway port.  I think what we actually want is the original source port from the request.

Reply:
hmm.. good question. Ok, correct me if I am wrong here but here is what I think: 

if we have a service such as: 

```
apiVersion: v1
kind: Service
metadata:
  name: some-service
spec:
  selector:
    app: some-service
  ports:
  - name: p1
    port: 8080
    targetPort: 8080
  - name: p2
    port: 8081
    targetPort: 8081
```

Then in this situation our target ports (this is the target port of the portPublisher type, not the endpoint) will e either 8080 or 8081. And this is ultimatelly what we want here. The real question however is, what happens if the port and the target port are different... like:

```
  - name: p1
    port: 8080
    targetPort: 9999
```

I think in this situation we want the port not the target port because... when this request hits the gateway we want to use this info to route to the service on the other cluster at 8080, not at 9999 (because there might not be 9999 at all). So I think I need to change that somehow to get the port not the targetPort. But to answer your original question, this will not be the port of the gateway. WDYT ?

Reply:
Yes, I think that's right.  But to put it more simply: I think we just want to preserve the port from the request.  In other words, the port (not the target port).

Reply:
yes, that is correct. Will change it to account for that.